orchagent/aclorch.h

```diff
@@ -168,6 +185,24 @@ class AclTableRangeMatch: public AclTableMatchInterface
 private:
     vector<int32_t> m_rangeList;
 };
+
+class MetaDataMgr
+{
+public:
+    void populateRange(uint16_t min, uint16_t max);
+    uint16_t getFreeMetaData(uint8_t dscp);
+    void recycleMetaData(uint16_t metadata);
+    bool isValidMetaData(uint16_t metadata);
+
+private:
+    bool initComplete = false;
+    uint16_t metaMin = 0;
+    uint16_t metaMax = 0;
+    list<uint16_t> m_freeMetadata;
+    map<uint8_t, uint16_t> m_dscpMetadata;
```

Comment:
pls check for uint32 for all references to metadata.

Reply:
i have limited the metadata to be between 0-4095. beyond this range, i double we ll have a platform which would allow 4k acl entries in a group.